### DISCUSSION TITLE
Lesson 7: Test basics

### DISCUSSION THREAD

**Discussion by CoggleChucker** [Cyfrin/foundry-full-course-cu#716] [Q&A] [answered]
Hi,
following are 3 tests I wrote for SimpleStorage.sol
```solidity
contract TestSimpleStorage is Test {
    SimpleStorage simpleStorage;

    function setUp() external {
        simpleStorage = new SimpleStorage();
    }

    function testDefaultValue() public {
        assertEq(simpleStorage.favoriteNumber(), 0);
    }

    function testFavoriteNumberChange() public {
        simpleStorage.SetFavoriteNumber(123);
        assertEq(simpleStorage.ShareFavoriteNumber(), 123);
    }

    function testCheckFavoriteNumberAgain() public {
        assertEq(simpleStorage.ShareFavoriteNumber(), 0);
    }
}
```
My question is,
do ```setUP()``` function runs only once, or it runs before each test(3 times in this case)?

**Comment by theirrationalone** [comment]
It's runs very first but once.... So last known state setup into `setUp` function would be true for each test... 

Let's we setup var  = 1 in `setUp` function.. 

In 1 test : var += 2 => 1+2 = 3

In 2  test : var += 10 => 1+ 10 = 11

In first test : var += 15 => 1+ 15 = 16
.
. 
. 
. 
. 
. 

In n test : var += 2 => 1+ n = n

**Reply by CoggleChucker** [reply]
so, does it reverts to its default state(or state set by ```setup()``` function) after each test?

**Reply by theirrationalone** [reply, marked as answer]
Yes of course, `setUp` runs once for each test but not for modifiers.

**Reply by CoggleChucker** [reply]
Thanks @theirrationalone

**Reply by theirrationalone** [reply]
You're Welcome @CoggleChucker ! ❤
